### PR TITLE
CNF-15645: Migrating Hardware Template ConfigMaps to CRDs

### PR DESCRIPTION
**Pull request by tliu2021**
This commit migrates the hardware template from ConfigMaps to CRDs. 
The following changes are included:
 1. Introduce the Hardware Template CRD: Defines the structure for representing hardware templates within CRDs.
 2. Update the NodePool CRD: Embeds NodePoolData to eliminate the need for redefining common fields (Name, HwProfile,  ResourcePoolID) in multiple places. This change improves code modularity, enhances reusability, and supports future extensibility.
 3. Remove the interface list from the NodePool.

### CODE REVIEW

**Comment by browsell** [comment]
Can't we just merge the plugin side changes at the same time to avoid the migration hop ?

**Comment by tliu2021** [comment]
> Can't we just merge the plugin side changes at the same time to avoid the migration hop ?

Yes, we can merge both changes simultaneously in a controlled manner.

**Review by Missxiaoguo — COMMENTED** [review]
Since it’s no longer a ConfigMap, the description for spec.templates.hwTemplate in the ClusterTemplate should be updated, as it currently says:
`// HwTemplate defines a reference to a hardware template configmap`.
Additionally, the [examples](https://github.com/openshift-kni/oran-o2ims/tree/main/docs/samples/git-setup/clustertemplates/hardwaretemplates/sno-ran-du) and the related section in the [docs](https://github.com/openshift-kni/oran-o2ims/blob/main/docs) should be updated to reflect the changes as well. 
Looks good to me otherwise.

**Comment by tliu2021** [comment]
> Since it’s no longer a ConfigMap, the description for spec.templates.hwTemplate in the ClusterTemplate should be updated, as it currently says: `// HwTemplate defines a reference to a hardware template configmap`. Additionally, the [examples](https://github.com/openshift-kni/oran-o2ims/tree/main/docs/samples/git-setup/clustertemplates/hardwaretemplates/sno-ran-du) and the related section in the [docs](https://github.com/openshift-kni/oran-o2ims/blob/main/docs) should be updated to reflect the changes as well. Looks good to me otherwise.

All references have been updated.

File: config/samples/v1alpha1_hardwaretemplate.yaml [hunk 1]
```diff
@@ -1,17 +1,25 @@
-apiVersion: v1
-kind: ConfigMap
+apiVersion: o2ims-hardwaremanagement.oran.openshift.io/v1alpha1
+kind: HardwareTemplate
 metadata:
   name: sample-hwtemplate-configmap-v1
   namespace: oran-o2ims
-data:
+spec:
```
[the remark this answers is not in the document]
Comment (tliu2021) [reply config/samples/v1alpha1_hardwaretemplate.yaml:6]: Yes, updated.

**Comment by Missxiaoguo** [comment]
One more is missing in the doc under the timeout configuration. The others look good to me!
https://github.com/openshift-kni/oran-o2ims/blob/main/docs/cluster-provisioning.md#timeout-configuration

**Comment by tliu2021** [comment]
> One more is missing in the doc under the timeout configuration. The others look good to me! https://github.com/openshift-kni/oran-o2ims/blob/main/docs/cluster-provisioning.md#timeout-configuration

Updated.

**Pull request merged**